woocommerce/code-reference
# New template with source code and faster search engine

This PR introduces a new template where many issues present in the default PHPDoc template gets solved, like missing source code, and the search that was crashing browsers.

Note that the new template is based in the latest template from PHPDoc, using the latest structure that makes everything more organized, and also keep us updated when PHPDoc 3 gets released.

And besides the new styling, I also included a hook reference generator based on the old one used for ApiGen. I'm waiting for a better documentation for PHPDoc 3, for now is missing a proper documentation to extend PHPDoc, so for now the hook generator should do the work.

Closes #6
Closes #7

### Steps to test:

If you want to just check how everything I said looks like, you can just a sample in: https://woocommerce.github.io/code-reference/v2/
But note that search the search uses links to https://woocommerce.github.io/code-reference/, so search results should redirect you to a different page.

Steps to build and test:

1. Download: `git clone https://github.com/woocommerce/code-reference.git --branch v2 --single-branch --depth 1`
2. Install dependencies: `cd code-reference/ && composer install`
3. Run build with latest WooCommerce release: `./deploy.sh -s 4.5.2 --build-only`. Make sure to use `--build-only` so you don't deploy this branch :laughing: 
4. Open the `build/api/index.html` in your browser: `firefox build/api/index.html`
